### PR TITLE
[graph] more tweaks to layout/reset

### LINKED ISSUE #7935: [graph] when zoomed-in "reset view" not resizing correctly
"Reset view" should perform a layout AND fit to the current screen. When zoomed-in The layout does not always fit inside the current screen.  Similar behavior can be seen in other situations, like graph hide.

For example:

[Screencast From 2024-11-18 21-34-15.webm](https://github.com/user-attachments/assets/e7ef22ac-6c89-48f2-9ce0-25a49c5705bd)

### PR DESCRIPTION
**Pull request by jshaughn**
Closes #7935 (again)

Increase the delay before performing a resize fit.  This seems to be enough in several tries to break it.

Also update react-resize-detector to the highest version prior to dropping support for react 17.

There are still times when PFT seems to just slightly lose its scale, I can't figure how to fix or avoid it.

To test you can just try various things, combining resize, changing layouts, zooming and panning, graph hide, side panel collapse/expand, etc.  Unless something horrible happens it's probably OK, even if it does sort of start laying out slightly too big.

### CODE REVIEW

**Comment by jshaughn** [comment]
> we can fit the graph twice...

Great idea.  Applied and pushed!

File: frontend/src/pages/GraphPF/GraphPF.tsx [hunk 1]
```diff
@@ -322,11 +322,13 @@ const TopologyContent: React.FC<{
     }
 
     if (layoutInProgress !== LayoutType.LayoutNoFit) {
+      controller.getGraph().fit(FIT_PADDING);
```
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/GraphPF.tsx:325]: This needs some refactoring. If the layout type is not Resize, you will fit the layout twice without any delay.

My suggestion (simply remove the else statement in the condition):

```
if (layoutInProgress !== LayoutType.LayoutNoFit) {
      controller.getGraph().fit(FIT_PADDING);

      // On a resize, delay fit to ensure that the canvas size updates before the fit
      if (layoutInProgress === LayoutType.Resize) {
        setTimeout(() => {
          controller.getGraph().fit(FIT_PADDING);
        }, 500);
      }
    }
```

Comment (jshaughn) [reply frontend/src/pages/GraphPF/GraphPF.tsx:325]: @ferhoyos OK, I cleaned it up.  I made one additional small tweak to the padding size, but I don't see anything else to do at this time.

File: frontend/src/pages/GraphPF/GraphPF.tsx [hunk 2]
```diff
@@ -886,6 +887,10 @@ const TopologyContent: React.FC<{
                   zoomOutCallback: () => {
                     controller && controller.getGraph().scaleBy(ZOOM_OUT);
                   },
+                  // currently unused
+                  fitToScreenCallback: () => {
+                    controller.getGraph().fit(80);
```
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/GraphPF.tsx:892]: Why do you hardcode 80 here? It should be the FIT_PADDING value right?

Comment (jshaughn) [reply frontend/src/pages/GraphPF/GraphPF.tsx:892]: good grief, I need a holiday, it seems.  That code isn't executed, but yes.  I'll push the change.

Comment (jshaughn) [reply frontend/src/pages/GraphPF/GraphPF.tsx]: done.

**Review by ferhoyos — APPROVED** [review]
LGTM

**Pull request merged**